### PR TITLE
[Windows] WiFi Direct API

### PR DESCRIPTION
**Pull request by astojilj**
Initial implementation of WiFi Direct experimental API on Windows.
Supported only on Windows10 and Android.
JS API is the same as in Android.

BUG=XWALK-5752

### CODE REVIEW

**Comment by astojilj** [comment]
@darktears @pozdnyakov 
Could you please help on reviewing this?

I'm not very happy with how usage of Universal API is resolved (Windows Kits\10\UnionMetadata\Windows.winmd reference in csproj file). Build and binary are working though on Windows 10 and Window8, though the feature is running only on Windows10.

Plan to add runtime feature flag to disable/enable the feature and update manual tests and documentation - this is the first working version.
Thanks.

File: experimental/wifidirect/XWalkExtensionInstance.cs [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+                removeConnected(jsonOutput);
+            }
+        }
+
+        public void HandleSyncMessage(String message) {
+            native_.SendSyncReply(message);
+        }
+
+        DeviceInformation findDiscoveredById(string id) {
+            for (int i = 0; i < mDiscovered.Count; i++) {
```
Comment (pozdnyakov) [inline experimental/wifidirect/XWalkExtensionInstance.cs]: nit: you could use `foreach`

File: experimental/wifidirect/XWalkExtensionInstance.cs [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+            mDeviceWatcher.Removed -= OnDeviceRemoved;
+            mDeviceWatcher.Updated -= OnDeviceUpdated;
+            mDeviceWatcher.Stopped -= OnStopped;
+            native_.PostMessageToJS(new JavaScriptSerializer().Serialize(createEventData(EVENT_DISCOVERY_STOPPED)));
+        }
+        #endregion DeviceWatcherEvents
+
+        #region Inviting
+        void connect(JSONObject response) {
+            JSONObject device = (JSONObject)response[TAG_DATA];
```
Comment (pozdnyakov) [inline experimental/wifidirect/XWalkExtensionInstance.cs]: won't `[]` implicitly create `device` instance?

**Comment by pozdnyakov** [comment]
lgtm

File: experimental/wifidirect/XWalkExtensionInstance.cs (hunk 2 above)
Comment (astojilj) [inline experimental/wifidirect/XWalkExtensionInstance.cs]: get raises KeyNotFound exception. now returning exception info to caller.

**Comment by astojilj** [comment]
added experimental runtime feature and KeyNotFound exception error handling.

**Pull request merged**